app/src/main/java/com/example/plantsapp/presentation/ui/plantdetail/PlantDetailViewModel.kt

```diff
@@ -16,9 +17,20 @@ class PlantDetailViewModel(
     private val _plant: MutableLiveData<Plant> = MutableLiveData()
     val plant: LiveData<Plant> get() = _plant
 
+    private val _toNavigateBack: MutableLiveData<Event<Unit>> = MutableLiveData()
+    val toNavigateBack: LiveData<Event<Unit>> get() = _toNavigateBack
+
     init {
         viewModelScope.launch {
             _plant.value = repository.getPlantByName(plantName)
         }
     }
+
+    fun onDelete(plant: Plant) {
+        viewModelScope.launch {
+            repository.deletePlant(plant)
+
+            _toNavigateBack.value = Event(Unit)
+        }
+    }
```

Comment:
You(probably) have plant inside of plant livedata - you can try to take it from here